src/pint/observatory/__init__.py

```diff
@@ -1,11 +1,29 @@
-"""Machinery to support PINT's list of observatories."""
+"""Machinery to support PINT's list of observatories.
+
+This code maintains a registry of observatories known to PINT.
+Observatories are added to the registry when the objects are created.
+For the observatories defined in PINT, these objects are created
+when the relevant module is imported.
+
+If you want to ensure that all PINT's observatories are available
+to your interpreter, you can run::
+
+    >>> import pint.observatory.observatories
+    >>> import pint.observatory.topo_obs
+    >>> import pint.observatory.special_locations
```

Comment:
I feel like this is a little clumsy.  It would be nice to offer a convenience function to list all loaded observatories after only a single import.  Like: `import pint.observatory;print(pint.observatory.names())`.  After the extra imports you can do `pint.observatory.Observatory.names()`: is there a better way?  A way that will capture satellites too?  Can we put those imports at the end of `__init__.py` without a circular import issue?

Reply:
maybe even `get_observatory()` without arguments will return the names.

Reply:
Constructing and registering all those objects at import time is an uncomfortable hack for just these kinds of reasons. I'm hesitant to make importing `pint.observatory` pull in all satellite and topocentric observatories, and of course any the user constructs only appear after the user has constructed them so we can't really create a `ensure_all_observatories_are_defined()` function (that would just do the above imports).

Reply:
I guess I am thinking of a user who wants to quickly figure out what a telescope is called (so including aliases could be helpful).  In astropy I can do:
```
import astropy.coordinates
astropy.coordinates.EarthLocation.get_site_names()
```
and that gives the names (although not aliases).  I do use that to e.g. figure out that I want `gemini_north` and not `gemini-north`.  

It looks like the function to update clock files does import all of the observatories (albeit in a different scope).  Is that not run regularly?

Reply:
.... actually `pint.observatory.Observatory.names()` is a function, I might as well do the imports in there, surely anyone calling that wants all of them? That and `get_observatory()` should cover all reasonable cases where people need the imports?

Reply:
This is a little annoying to test as all test cases are run in the same python session, so the imports will basically always have happened by the time the test is run. It's actually the notebooks that are more likely to catch this sort of thing.

Reply:
Don't forget about `get_satellite_observatory()`

Reply:
You can't know the names of satellite observatories in advance. They are defined at runtime based on an orbit file.

Reply:
So these are observatories that don't become available until the user loads an orbit file? I can do the import, but they won't appear in `names()` until that happens.

Reply:
Maybe this could also wait until observatories -> data file change happens, at which point it wouldn't be a python import but reading a data-file.